cmake/tests/cxx14_auto.cpp

```diff
@@ -0,0 +1,20 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2009 Andrey Semashev
+//  Copyright (c) 2015 Agustin Berge
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+template<typename T>
+auto check_f(T& i)
+{
+    T j = 1 + j;
+    return j;
+}
```

Comment:
Ah, got it. You're right, this is apparently a bug in NVCC. However, CMake isn't using NVCC to test for compiler flags (but rather the default C++ compiler), so a test here for that specific defect likely wouldn't yield us much.

If you object to this test as it is in the PR then I think it would be fine to just tell CUDA users to manually set "-DWITH_CXX11_AUTO=false". In that case I'd remove the new C++14 auto test. Your thoughts?